Modules/SubModule_Import/xASL_imp_NII2BIDS_Session.m

```diff
@@ -0,0 +1,71 @@
+function xASL_imp_NII2BIDS_Session(imPar, bidsPar, studyPar, listSessions, nameSubjectSession, bidsLabel, iSession)
+%xASL_imp_NII2BIDS_SubjectSession NII2BIDS conversion for a single sessions.
+%
+% FORMAT: xASL_imp_NII2BIDS_SubjectSession(imPar, bidsPar, studyPar, listSessions, nameSubjectSession, subjectLabel, iSession)
+% 
+% INPUT:
+%   imPar                 - JSON file with structure with import parameter (STRUCT, REQUIRED)
+%   bidsPar               - Output of xASL_imp_Config (STRUCT, REQUIRED)
+%   studyPar              - JSON file with the BIDS parameters relevant for the whole study (STRUCT, REQUIRED)
+%   listSessions          - list of sessions (CELL ARRAY, REQUIRED, e.g.: {'ASL_1'})
+%   nameSubjectSession    - name of the subject (CELL ARRAY, REQUIRED)
+%   subjectLabel          - subject label (CHAR ARRAY, REQUIRED)
+%   iSession              - Session number (INTEGER, REQUIRED)
+%
+% OUTPUT:
+%  n/a
+%                         
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: NII2BIDS conversion for a single sessions.
+% 
+% 1. Make a subject directory
+% 2. Iterate over runs
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:     n/a
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    %% Unpack bidsLabel struct
+    subjectLabel = bidsLabel.subject;
+    visitLabel = bidsLabel.visit;
+
+    %% 1. Make a subject directory
+    if length(listSessions)>1 || ~isempty(visitLabel)
+        if ~isempty(visitLabel)
+            sessionLabel = ['ses-' visitLabel];
+        else
+            sessionLabel = ['ses-' listSessions{iSession}(5:end)];
+        end
+
+        if ~exist(fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel),'dir')
+            xASL_adm_CreateDir(fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel));
+            xASL_adm_CreateDir(fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel,'perf'));
+        end
+        inSessionPath = fullfile(imPar.TempRoot,nameSubjectSession,listSessions{iSession});
+        outSessionPath = fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel);
```

Comment:
Adding `sub-` as suffix is fine as long as we check if this is not double (in case this suffix `sub-` was already in the DICOM folder). Same for `ses-`.

Reply:
That's true, but DCM2NII & NII2BIDS heavily depend on the sourceStructure.json, where we define the tokens. With the tokens we controlled that the subject ids and session ids are only the id without the prefix so far. I didn't change that part though, like the other code this was only moved from the original xASL_imp_NII2BIDS_SubjectSession.

Reply:
Sure, but we don't get to see what the users do. So we only have to add a check later that it doesn't have twice `sub-` in the name, same for `ses-`.

Reply:
Good point - let's make a new issue for that. Or rather lets glue that to #775 (I've just done that).